tests/api/generate_address_test.rs

```diff
@@ -21,8 +70,10 @@ async fn generate_address_returns_valid_json_data_test() {
         .await
         .expect("Failed to execute request");
 
+    let resp_body = response.json::<GenerateAddressResponse>().await.unwrap();
+
     // 3. Assert
-    assert!(response.status().is_success());
+    assert_eq!("Address generated successfully", resp_body.message);
```

Comment:
I would also advise adding the assertion for status code and maybe the expected length of the address returned